vrecord

```diff
@@ -131,6 +130,25 @@ _check_mpv(){
     fi
 }
 
+# Set variables for system differences
+QUERY_OS_TYPE="$(uname -s)"
+if [[ "${QUERY_OS_TYPE}" = "Darwin" ]] ; then
+    OS_TYPE="macOS"
+    DEFAULTFONT="/Library/Fonts/Andale Mono.ttf"
+    HARDWARE_REPORT=$(system_profiler SPHardwareDataType)
+    CORE_COUNT="$(_parse_report "Total Number of Cores" "${HARDWARE_REPORT}")"
+    OPEN_COMMAND="open"
+elif [[ "${QUERY_OS_TYPE}" = "Linux" ]] ; then
+    OS_TYPE="linux"
+    DEFAULTFONT="/usr/share/fonts/truetype/liberation/LiberationSans-Regular.ttf"
+    HARDWARE_REPORT=$(lscpu)
+    CORE_COUNT="$(nproc --all)"
+    OPEN_COMMAND="xdg-open"
+else
+    echo "Unsupported OS, ${QUERY_OS_TYPE}, detected. Exiting."
```

Comment:
To me, that way it would be easier to read: 
`echo "Unsupported OS detected: ${QUERY_OS_TYPE}. Exiting."`